### PR TITLE
Remove checking of sync commit ids

### PR DESCRIPTION
**Pull request by arteam**
A Lucene commit doesn't contain sync ids `SegmentInfos` anymore, so we can't rely on them during recovery. The fields was marked as deprecated in #102343.

### CODE REVIEW

File: server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java [hunk 1]
```diff
@@ -432,4 +432,10 @@ Table buildTable(RestRequest request, ClusterStateResponse state, IndicesStatsRe
 
         return table;
     }
+
+    // Remove sync_id from the API in V10?
+    @UpdateForV10(owner = UpdateForV10.Owner.DISTRIBUTED_INDEXING)
+    private static void addSyncId(Table table) {
```
Comment (arteam) [inline server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java:438]: We can remove this field from the API output without making a breaking change. Does it make sense to return conditionally only for 8.x based on `RestVersion`?

**Comment by arteam** [comment]
@elasticmachine update branch

**Review by kingherc — COMMENTED** [review]
Looking good! A couple of comments.

File: server/src/test/java/org/elasticsearch/index/store/StoreTests.java [hunk 2]
```diff
@@ -997,16 +997,13 @@ public void testUserDataRead() throws IOException {
         doc.add(new TextField("id", "1", Field.Store.NO));
         writer.addDocument(doc);
         Map<String, String> commitData = Maps.newMapWithExpectedSize(2);
-        String syncId = "a sync id";
-        commitData.put(Engine.SYNC_COMMIT_ID, syncId);
         writer.setLiveCommitData(commitData.entrySet());
```
[the remark this answers is not in the document]
Comment (arteam) [reply server/src/test/java/org/elasticsearch/index/store/StoreTests.java]: Done!

File: server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java [hunk 3]
```diff
@@ -320,7 +319,8 @@ Table buildTable(RestRequest request, ClusterStateResponse state, IndicesStatsRe
                 table.addCell(null);
             }
 
-            table.addCell(commitStats == null ? null : commitStats.getUserData().get(Engine.SYNC_COMMIT_ID));
+            // Remove sync_id from the API?
+            table.addCell(null);
```
[the remark this answers is not in the document]
Comment (arteam) [reply server/src/main/java/org/elasticsearch/rest/action/cat/RestShardsAction.java:323]: Right, I originally did that, caused quite a few issues with the BWC tests, but I'm gonna take another stab :+1:

**Comment by arteam** [comment]
@elasticmachine update branch

File: server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java [hunk 4]
```diff
@@ -1333,25 +1331,16 @@ public void testSyncedFlushVanishesOnReplay() throws IOException {
...
         engine.close();
         engine = new InternalEngine(config);
         recoverFromTranslog(engine, translogHandler, Long.MAX_VALUE);
-        assertNull(
-            "Sync ID must be gone since we have a document to replay",
-            engine.getLastCommittedSegmentInfos().getUserData().get(Engine.SYNC_COMMIT_ID)
-        );
     }
 
     void syncFlush(IndexWriter writer, InternalEngine engine, String syncId) throws IOException {
```
Comment (kingherc) [inline server/src/test/java/org/elasticsearch/index/engine/InternalEngineTests.java:1349]: Lets remove the syncId argument as well

**Review by kingherc — COMMENTED** [review]
Looks good so far. Find some more comments. And also some comments/questions on places I've found "sync_id" mentions somehow, and would like us to consider whether they need removing/adapting:

* `ShardStoreInfo#hasMatchingSyncId` seems relevant. Could you check all code paths to the creation of ShardStoreInfo, whether they may give Long.MAX_VALUE matchingBytes ever? If they never give it, we could remove it.
* ReplicaShardAllocatorIT.java mentions a "sync_id" in a comment, please see to it (remove or adapt it).

File: rest-api-spec/build.gradle [hunk 5]
```diff
@@ -63,4 +63,5 @@ tasks.named("yamlRestCompatTestTransform").configure ({ task ->
   task.skipTest("cat.aliases/10_basic/Deprecated local parameter", "CAT APIs not covered by compatibility policy")
   task.skipTest("cluster.desired_nodes/10_basic/Test delete desired nodes with node_version generates a warning", "node_version warning is removed in 9.0")
   task.skipTest("cluster.desired_nodes/10_basic/Test update desired nodes with node_version generates a warning", "node_version warning is removed in 9.0")
+  task.skipTest("cat.shards/10_basic/Help", "sync_id is removed in 9.0")
```
[the remark this answers is not in the document]
Comment (arteam) [reply rest-api-spec/build.gradle:68]: I wasn't sure that `gte_v9.0.0` is actually supported, but let me try that

**Review by kingherc — APPROVED** [review]
LGTM, thanks for the changes!

**Pull request merged**